// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// This API is experimental and under-development.

syntax = "proto3";

package sax.server.audio;

import "saxml/protobuf/common.proto";

message AsrRequest {
  enum AudioFormat {
    UNSPECIFIED = 0;
    WAVEFORM_FILE = 1;   // Raw waveform file.
    DATA_UTTERANCE = 2;  // speech.DataUtterance proto.
  }

  string model_key = 1;
  bytes audio_bytes = 2;
  AudioFormat audio_format = 3;  // TODO(sax-dev): Move to extra_inputs.
  .sax.ExtraInputs extra_inputs = 4;
}

message AsrHypothesis {
  // Recognized text and correcponding score.
  string text = 1;
  double score = 2;
}

message AsrResponse {
  // Top hyps ordered by score in descending order.
  repeated AsrHypothesis hyps = 1;
}

service AudioService {
  // Returns the recognized text given the audio.
  rpc Recognize(AsrRequest) returns (AsrResponse);
}
